python/paddle/fluid/tests/unittests/test_eigvalsh_op.py

```diff
@@ -21,6 +21,31 @@
 from gradient_checker import grad_check
 
 
+def compare_result(actual, expected):
+    assert actual.ndim == 1 or actual.ndim == 2
+
+    if actual.ndim == 1:
+        valid_eigenvalues(actual, expected)
+        return
+
+    for batch_actual, batch_expected in zip(actual, expected):
+        valid_eigenvalues(batch_actual, batch_expected)
+
+
+def valid_eigenvalues(actual, expected):
+
+    FP32_MAX_RELATIVE_ERR = 5e-5
+    FP64_MAX_RELATIVE_ERR = 1e-14
+
+    rtol = FP32_MAX_RELATIVE_ERR if actual.dtype == np.single else FP64_MAX_RELATIVE_ERR
+
+    diff = np.abs(expected - actual)
+    max_diff = np.max(diff)
+    max_ref = np.max(expected)
+    relative_error = max_diff / max_ref
```

Comment:
max_diff and max_ref may not be derived from the same element of eigenvalues, why not use np.allclose to check rtol and atol of all elements

Reply:
cusolver valid the eigenvalues by ensuring `|| d_ref - d_cusolver || / max|d_ref|` is less than relative error. The `norm` here is to get the max values of eigenvalues. All elements in `diff` are expected to be closed to 0 and also proportional to the input(output) scale, that's why we have to compare the `max(abs(expected))`.

So `max_ref = np.max(expected)` should be `np.max(np.abs(expected))`, let me update it.

Reply:
> proportional to the input(output) scale

It means the max residual becomes larger when the elements in input(output) get larger.

Reply:
theoritically, np.allclose(expected, actual, rtol, atol) is more reasonable? 
if max_diff comes from eigenvalue e1 and max_ref from eigenvalue e2, I do not think it's reasonable

Reply:
As I said, the formula here is to implement `|| d_ref - d_cusolver || / max|d_ref| < tolerance` to align cusolver validation. If we want to choose a difference validation method, there is the risk to set a difference tolerance across different cuda version (https://github.com/PaddlePaddle/Paddle/issues/39446).

Reply:
by reading issue, I understand backgroud, thank you.
but I have another question: does this increase in error mean that the calculation is more accurate or less

Reply:
It is less accurate if we see the **third-party library as the ground truth**. But if we calculate the error through [eigendecomposition](https://en.wikipedia.org/wiki/Eigendecomposition_of_a_matrix), the error is in the same order and acceptable by cusolver team.

Reply:
ok, thank you